constraint/pkg/client/drivers/local/driver.go

```diff
@@ -290,14 +314,37 @@ func (d *Driver) Query(ctx context.Context, target string, constraints []*unstru
 			return nil, nil, err
 		}
 
-		for _, result := range kindResults {
-			result.EvaluationMeta = RegoEvaluationMeta{
-				TemplateRunTime: float64(evalEndTime.Nanoseconds()) / 1000000,
-				ConstraintCount: uint(len(kindResults)),
+		results = append(results, kindResults...)
+
+		if d.gatherMetrics {
+			// for the batch of evaluated constraints, report their evaluation metadata
+			evalRunTime := float64(evalEndTime.Nanoseconds()) / 1000000
+
+			for _, k := range kindConstraints {
+				statsEntry := &instrumentation.StatsEntry{
+					Key: drivers.ConstraintKeyFrom(k),
+					AllStats: []*instrumentation.NamedStats{
+						{
+							StatsName: "EvaluationMetadata",
+							Stats: []*instrumentation.Stat{
+								{
+									Name:        "templateRunTimeMillis",
+									Value:       fmt.Sprintf(" %.2f", evalRunTime),
+									Description: "the number of milliseconds it took to evaluate all constraints for a template",
+								},
+								{
+									Name:        "engineType",
```

Comment:
I like the idea of reporting the engine type. This is an example of structure that we can enforce by being intentional about how we collate the responses to multiple calls to Query() (likely in client.go)

Reply:
this was a good call out! I was focused more on providing data for every `kind, name` tuple for constraints per the result backfilling mindset. Reworked this for `kind` level stats only.